projects/igniteui-angular/src/lib/services/overlay/scroll/AbsoluteScrollStrategy.ts

```diff
@@ -0,0 +1,70 @@
+import { IScrollStrategy } from './IScrollStrategy';
+import { IgxOverlayService } from '../overlay';
+import { IPositionStrategy } from '../position';
+import { ElementRef, ComponentRef } from '@angular/core';
+import { OverlaySettings, PositionSettings, Point } from '../utilities';
+
+export class AbsoluteScrollStrategy implements IScrollStrategy {
+    private _initialized = false;
+    private _document: Document;
+    private _overlayService: IgxOverlayService;
+    private _id: string;
+    private _scrollContainer: HTMLElement;
+    private _component: {
+        id: string,
+        elementRef: ElementRef,
+        componentRef: ComponentRef<{}>,
+        size: { width: number, height: number },
+        settings: OverlaySettings
+    };
+
+    constructor(scrollContainer?: HTMLElement) {
+        this._scrollContainer = scrollContainer;
+    }
+
+    initialize(document: Document, overlayService: IgxOverlayService, id: string) {
```

Comment:
How about we simplify the `IScrollStrategy` interface (without `initialize`, you can pass in the service + id in the constructor if needed).
Plus add a `[re]position(id: string)` method to the service to remove all the position settings being passed around. The scroll strategy should be completely decoupled from that.

Reply:
Reposition added to overlay.
We need reference to document, as we attach some events to it. As we cannot inject it through the constructor we are passing it via initialize method.